espressif-devcon/2023/2023.md

```diff
@@ -17,14 +17,178 @@ headingDivider: 1
 
 Last year's talk: [EDC22 Day 1 Talk 7: Rust on Espressif chips](https://www.youtube.com/watch?v=qeEmJ_-6fPg).
 
-`async` in Rust adds two new keywords to the language, `async` & `await`, where 
+`async` in Rust adds two new keywords to the language, `async` & `await`, where
 
  - `async` defines a block or function to be asynchronous
  - `await` defines yield points _within_ an `async` block or function.
 
+## async/.await
 
-## Why use async?
+- building blocks for async blocks, closures, and functions that can be paused and yield control back to the caller
+- `async` defines blocks, closures, and functions to be asynchronous
+- `await` defines a yield point within `async` code
+- `async` code returns a value that implements the `Future` trait
 
+```rust
+pub async fn say_hello(uart0: &mut Serial<UART0>) {
+  let message = "Hello World!";
+  uart0.write_bytes(message).await; // yield point here!
+  let message = "Goodbye";
+  uart0.write_bytes(message).await; // another yield point here!
+}
+```
+
+## The Future Trait
```

Comment:
I think we can remove this section, up until the "More practical example" part. I go into this detail in the how it works section. I'm also not sure why we are simplifying the Futurer trait? Why don't we just copy it as it?